engine/Assets/Scripts/UI/Dynamic/Modals/Configuring/ConfigMotorModal.cs

```diff
@@ -0,0 +1,261 @@
+            _robot.GetLeftRightWheels()!.Value.leftWheels.ForEach(x => {
+                _driveMotors[i]        = new ConfigMotor();
+                _driveMotors[i].driver = x;
+                _driveDrivers[i]       = x;
+                i++;
+            });
+            _robot.GetLeftRightWheels()!.Value.rightWheels.ForEach(x => {
+                _driveMotors[i]        = new ConfigMotor();
+                _driveMotors[i].driver = x;
+                _driveDrivers[i]       = x;
+                i++;
+            });
+        }
+        
+        //grab original target velocity for drivetrain
+        motorCount             += SimulationManager.Drivers[_robot.Name].Count - _driveMotors.Length;
+        _driveMotors[0].origVel = (_driveMotors[0].driver as WheelDriver).Motor.targetVelocity;
+        if (_robotISSwerve) {
+            _turnMotors[0].origVel = (_turnMotors[0].driver as RotationalDriver).Motor.targetVelocity;
+        }
+
+        Driver[] drivers = new Driver[SimulationManager.Drivers[_robot.Name].Count];
+        i = 0;
+        SimulationManager.Drivers[_robot.Name].ForEach(x => {
+            drivers[i] = x;
+        });
+
+        //set up other motors array
+        _otherMotors = new ConfigMotor[motorCount];
+        i = 0;
+        SimulationManager.Drivers[_robot.Name].ForEach(x => {
+            if (Array.IndexOf(_driveDrivers, x) == -1 && (!_robotISSwerve || Array.IndexOf(_turnDrivers, x) == -1)) {
+                _otherMotors[i]        = new ConfigMotor();
+                _otherMotors[i].driver = x;
+                i++;
+            }
+        });
+        
+        //UI
+        Title.SetText("Motor Configuration").SetWidth<Label>(MODAL_WIDTH - PADDING * 4);
+        Description.SetText("Change motor settings");
+
+        AcceptButton.StepIntoLabel(b => { b.SetText("Save"); }).AddOnClickedEvent(b => {
+            RobotSimObject.GetCurrentlyPossessedRobot().MiraLive.Save();
+            DynamicUIManager.CloseActiveModal();
+            
+            //Save to Mira
+            if (_driveMotors[0].velChanged) {
+                _driveMotors.ForEach(x => {
+                    SaveToMira(x.driver as WheelDriver);
+                });
+            }
+            if (_robotISSwerve && _turnMotors[0].velChanged) {
+                _turnMotors.ForEach(x => {
+                    SaveToMira(x.driver as RotationalDriver);
+                });
+            }
+            _otherMotors.ForEach(x => {
+                if (x.velChanged) {
+                    if (x.driver is WheelDriver) {
+                        SaveToMira(x.driver as WheelDriver);
+                    } else if (x.driver is RotationalDriver) {
+                        SaveToMira(x.driver as RotationalDriver);
+                    }
+
+                }
+            });
+        });
+
+        MiddleButton.SetWidth<Button>(132)
+            .SetAnchor<Button>(new Vector2(0.55f, 0), new Vector2(0.55f, 0))
```

Comment:
Leaving it for now until UI changes, but quick clean changes are appreciated if you have ideas.